From d538b48584d0d20b2a3e74bbfc32ce7b33c14d2c Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:31 -0400
Subject: [scsi] qla2xxx: Wait for reset completion without lock for ISPFX00

Message-id: <1408391817-6389-44-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87932
O-Subject: [PATCH RHEL 7.1 e-stor 43/69] qla2xxx: Wait for reset completion without lock for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Saurav Kashyap <saurav.kashyap@qlogic.com>

scsi commit id ec1937a21ba193b489f669484e2180b9e08f0c7f

Signed-off-by: Giridhar Malavali <giridhar.malavali@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 84c038e..9676630 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -572,12 +572,13 @@ qlafx00_soc_cpu_reset(scsi_qla_host_t *vha)
  /* Kick in Core0 to start boot process */
  QLAFX00_SET_HBA_SOC_REG(ha, SOC_SW_RST_CONTROL_REG_CORE0, (0xF00));
 
+ spin_unlock_irqrestore(&ha->hardware_lock, flags);
+
  /* Wait 10secs for soft-reset to complete. */
  for (cnt = 10; cnt; cnt--) {
   msleep(1000);
   barrier();
  }
- spin_unlock_irqrestore(&ha->hardware_lock, flags);
 }
 
 /**
-- 
1.7.1